core/src/main/java/org/verapdf/features/objects/AnnotationFeaturesObject.java

```diff
@@ -0,0 +1,178 @@
+	}
+
+	/**
+	 * Reports all features from the object into the collection
+	 *
+	 * @return FeatureTreeNode class which represents a root node of the
+	 * constructed collection tree
+	 * @throws FeatureParsingException occurs when wrong features tree node constructs
+	 */
+	@Override
+	public FeatureTreeNode collectFeatures() throws FeatureParsingException {
+		AnnotationFeaturesObjectAdapter annotationAdapter = (AnnotationFeaturesObjectAdapter) this.adapter;
+		FeatureTreeNode root = FeatureTreeNode.createRootNode(ANNOTATION);
+		String id = annotationAdapter.getId();
+		if (id != null) {
+			root.setAttribute(CreateNodeHelper.ID, id);
+		}
+
+		CreateNodeHelper.addNotEmptyNode(SUB_TYPE, annotationAdapter.getSubtype(), root);
+		CreateNodeHelper.addBoxFeature("rectangle", annotationAdapter.getRectangle(), root);
+		CreateNodeHelper.addNotEmptyNode(CONTENTS, annotationAdapter.getContents(), root);
+		CreateNodeHelper.addNotEmptyNode(ANNOTATION_NAME, annotationAdapter.getAnnotationName(), root);
+		CreateNodeHelper.addNotEmptyNode(MODIFIED_DATE, annotationAdapter.getModifiedDate(), root);
+
+		Set<String> formXObjects = annotationAdapter.getFormXObjectsResources();
+
+		if (formXObjects != null && !formXObjects.isEmpty()) {
+			FeatureTreeNode resources = root.addChild("resources");
+			for (String xObjID : formXObjects) {
+				if (xObjID != null) {
+					FeatureTreeNode xObjNode = resources.addChild("xobject");
+					xObjNode.setAttribute(CreateNodeHelper.ID, xObjID);
+				}
+			}
+		}
+
+		String popupId = annotationAdapter.getPopupId();
+		if (popupId != null) {
+			FeatureTreeNode popup = root.addChild("popup");
+			popup.setAttribute(CreateNodeHelper.ID, popupId);
+		}
+
+		CreateNodeHelper.addDeviceColorSpaceNode("color", annotationAdapter.getColor(), root, this);
+
+		CreateNodeHelper.addNotEmptyNode(INVISIBLE, String.valueOf(annotationAdapter.isInvisible()), root);
+		CreateNodeHelper.addNotEmptyNode(HIDDEN, String.valueOf(annotationAdapter.isHidden()), root);
+		CreateNodeHelper.addNotEmptyNode(PRINT, String.valueOf(annotationAdapter.isPrinted()), root);
+		CreateNodeHelper.addNotEmptyNode(NO_ZOOM, String.valueOf(annotationAdapter.isNoZoom()), root);
+		CreateNodeHelper.addNotEmptyNode(NO_ROTATE, String.valueOf(annotationAdapter.isNoRotate()), root);
+		CreateNodeHelper.addNotEmptyNode(NO_VIEW, String.valueOf(annotationAdapter.isNoView()), root);
+		CreateNodeHelper.addNotEmptyNode(READ_ONLY, String.valueOf(annotationAdapter.isReadOnly()), root);
+		CreateNodeHelper.addNotEmptyNode(LOCKED, String.valueOf(annotationAdapter.isLocked()), root);
+		CreateNodeHelper.addNotEmptyNode(TOGGLE_NO_VIEW, String.valueOf(annotationAdapter.isToggleNoView()), root);
+		CreateNodeHelper.addNotEmptyNode(LOCKED_CONTENTS, String.valueOf(annotationAdapter.isLockedContents()), root);
+
+		return root;
+	}
+
+	/**
+	 * @return null
+	 */
+	@Override
+	public FeaturesData getData() {
+		return null;
+	}
+
+	static List<Feature> getFeaturesList() {
```

Comment:
There are a few places where you're falling back on default scope, like here. It might be better if this was explicitly scoped rather than implicitly.